framework/auth/core.py

```diff
@@ -667,6 +667,7 @@ def add_unclaimed_record(self, node, referrer, given_name, email=None):
             'email': clean_email,
         }
         self.unclaimed_records[project_id] = record
+        self.save()
```

Comment:
In order to avoid implicit saving, it might be better to remove this and call save after `add_unclaimed_record` is called.